### PR TITLE
Update Azure specifics

### PR DESCRIPTION
**Pull request by ncdc**
Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

### CODE REVIEW

**Comment by ncdc** [comment]
cc @paulbouwer - this incorporates most of your docs for installing on Azure.

File: docs/cloud-provider-specifics.md [hunk 1]
```diff
@@ -181,75 +185,108 @@ Now that you have your Google Cloud credentials stored in a Secret, you need to
...
     ```
 
 2. There are seven environment variables that need to be set for Heptio Ark to work properly. The following steps detail how to acquire these, in the process of setting up the necessary RBAC.
 
-3. List your account:
+3. Obtain your Azure Account Subscription ID and Tenant ID:
 
+    ```bash
+    AZURE_SUBSCRIPTION_ID=`az account list --query '[?isDefault].id' -o tsv`
+    AZURE_TENANT_ID=`az account list --query '[?isDefault].tenantId' -o tsv`
```
Comment (nrb) [inline docs/cloud-provider-specifics.md]: +1 to having specific commands for getting this info.

File: docs/cloud-provider-specifics.md [hunk 2]
```diff
@@ -267,6 +304,34 @@ Now that you have your Azure credentials stored in a Secret, you need to replace
 
   * Replace `<YOUR_BUCKET>`, `<YOUR_LOCATION>`, and `<YOUR_TIMEOUT>`. See the [Config definition][8] for details.
 
+Here is an example of a completed file.
```
Comment (nrb) [inline docs/cloud-provider-specifics.md]: +1 for complete examples

**Comment by ncdc** [comment]
cc @jimzim @rasquill @squillace - would appreciate any feedback asap if you have time!

File: docs/cloud-provider-specifics.md [hunk 3]
```diff
@@ -181,75 +185,108 @@ Now that you have your Google Cloud credentials stored in a Secret, you need to
...
     ```
-    az storage account create \
-        --name $AZURE_STORAGE_ACCOUNT_ID \
-        --resource-group $AZURE_RESOURCE_GROUP \
-        --sku Standard_GRS
-    ```
-    You will encounter an error message if the storage account ID is not unique; change it accordingly.
 
-8. Get the keys for your storage account:
+6. Create the storage account and blob container for Ark to store the backups in.
```
Comment (skriss) [inline docs/cloud-provider-specifics.md]: We don't have a section for creating a bucket for AWS/GCP. Would be nice to be consistent across the 3.

File: docs/cloud-provider-specifics.md [hunk 4]
```diff
@@ -181,75 +185,108 @@ Now that you have your Google Cloud credentials stored in a Secret, you need to
...
 
+#### Kubernetes cluster prerequisites
+
+Ensure that the VMs for your agent pool allow Managed Disks. If I/O performance is critical,
+consider using Premium Managed Disks, as these are SSD backed.
+
 #### Service principal creation
 To integrate Heptio Ark with Azure, you should follow the instructions below to create an Ark-specific [service principal][17].
 
 1. If you do not have the `az` Azure CLI 2.0 locally installed, follow the [user guide][18] to set it up. Once done, run:
```
Comment (paulbouwer) [inline docs/cloud-provider-specifics.md]: The correct install guide for Azure CLI 2.0 is located at: https://docs.microsoft.com/en-us/cli/azure/install-azure-cli.

I would modify this line as follows:

> follow the [install guide](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) to set it up.

File: docs/cloud-provider-specifics.md [hunk 5]
```diff
@@ -181,75 +185,108 @@ Now that you have your Google Cloud credentials stored in a Secret, you need to
...
 
+    ```bash
+    az group list --query '[].{ ResourceGroup: name, Location:location }'
     ```
-    az ad sp create-for-rbac --role="Contributor" --name="heptio-ark"
-    ```
-    From the response, save `appId` into `$AZURE_CLIENT_ID` and `password` into `$AZURE_CLIENT_SECRET`.
 
-6. Login into the `heptio-ark` service principal account:
+    Get your cluster's group `name` from the response, and use it to set `$AZURE_RESOURCE_GROUP`. (Also note the `location`--this is later used in the Azure-specific portion of the Ark Config).
```
Comment (paulbouwer) [inline docs/cloud-provider-specifics.md]: Get your cluster's Resource Group name from the `ResourceGroup` value in the response, and use it to set `$AZURE_RESOURCE_GROUP`. (Also note the `Location` value in the response - this is later used in the Azure-specific portion of the Ark Config).

**Comment by paulbouwer** [comment]
@ncdc @skriss @nrb - I added a few minor comments, but otherwise LGTM.

I built the updated guidance from a few passes at installing Ark on Azure, so the commands have been tested and should all **just work<sup>TM</sup>**.

File: docs/cloud-provider-specifics.md (hunk 3 above)
Comment (ncdc) [reply docs/cloud-provider-specifics.md]: I'll file issues

**Comment by ncdc** [comment]
@paulbouwer thanks! I've made the updates you suggested.

**Comment by nrb** [comment]
LGTM, merging.

**Pull request merged**